drwmutex.go

```diff
@@ -379,61 +375,28 @@ func (dm *DRWMutex) ForceUnlock() {
 
 // sendRelease sends a release message to a node that previously granted a lock
 func sendRelease(c NetLocker, name, uid string, isReadLock bool) {
-
-	backOffArray := []time.Duration{
-		30 * time.Second, // 30 secs
-		1 * time.Minute,  // 1 min
-		3 * time.Minute,  // 3 min
-		10 * time.Minute, // 10 min
-		30 * time.Minute, // 30 min
-		1 * time.Hour,    // 1 hour
+	// All client methods issuing RPCs are thread-safe and goroutine-safe,
+	// i.e. it is safe to call them from multiple concurrently running goroutines.
+	args := LockArgs{
+		UID:             uid,
+		Resource:        name,
+		ServerAddr:      clnts[ownNode].ServerAddr(),
+		ServiceEndpoint: clnts[ownNode].ServiceEndpoint(),
 	}
 
-	go func(c NetLocker, name string) {
-
-		for _, backOff := range backOffArray {
-
-			// All client methods issuing RPCs are thread-safe and goroutine-safe,
-			// i.e. it is safe to call them from multiple concurrently running goroutines.
-			args := LockArgs{
-				UID:             uid,
-				Resource:        name,
-				ServerAddr:      clnts[ownNode].ServerAddr(),
-				ServiceEndpoint: clnts[ownNode].ServiceEndpoint(),
-			}
-
-			var err error
-			if len(uid) == 0 {
-				if _, err = c.ForceUnlock(args); err != nil {
-					log("Unable to call ForceUnlock", err)
-				}
-			} else if isReadLock {
-				if _, err = c.RUnlock(args); err != nil {
-					log("Unable to call RUnlock", err)
-				}
-			} else {
-				if _, err = c.Unlock(args); err != nil {
-					log("Unable to call Unlock", err)
-				}
-			}
-
-			if err != nil {
-				// Ignore if err is net.Error and it is occurred due to timeout.
-				// The cause could have been server timestamp mismatch or server may have restarted.
-				// FIXME: This is minio specific behaviour and we would need a way to make it generically.
-				if nErr, ok := err.(net.Error); ok && nErr.Timeout() {
-					err = nil
-				}
-			}
-
-			if err == nil {
-				return
-			}
-
-			// Wait..
-			time.Sleep(backOff)
+	if len(uid) == 0 {
```

Comment:
With the lock maintenance that is implemented now by checking back with the original server whether the lock is still alive, there is no longer a need to try and deliver the release message at "all costs" (which even the previously implemented back-off mechanism could not guarantee).

So we can just send the release message once and, in case of non-delivery of the release message, rely on the lock maintenance algorithm to later on release the stale lock.